src/test/run-pass/ifmt.rs

```diff
@@ -246,3 +248,96 @@ fn test_order() {
                        foo(), foo(), foo(), a=foo(), b=foo(), c=foo()),
                "1 2 4 5 3 6".to_string());
 }
+
+fn test_collections() {
```

Comment:
I think this would be better placed in libcollectionstest. I think this file is mostly just verifying that the `format` macro itself works right.